tiled/client/utils.py

```diff
@@ -143,25 +162,37 @@ def export_util(file, format, get, link, params):
             format = ".".join(
                 suffix[1:] for suffix in Path(file).suffixes
             )  # e.g. "csv"
-        content = handle_error(
-            get(
-                link,
-                params={**parse_qs(urlparse(link).query), "format": format, **params},
-            )
-        ).read()
+        for attempt in retry_context():
+            with attempt:
```

Comment:
I have only seen the landing page for stamina. Would it be better to bake retry as a function decorator to `handle_error`? Removes lots of nesting complexity every time you have a request.

Reply:
Great question. I explored some options because, as you'd imagine, I was reluctant to do a bunch of tedious edits throughout `tiled.client`.

It comes down to this observation from the stamina docs:

> Sometimes you only want to retry a part of a function. Since iterators can’t catch exceptions and context managers can’t execute the same block multiple times, we need to combine them to achieve that.

Reply:
I did intentionally wrap `stamina.retry_context` in our own `retry_context` so that we can tweak the settings in one place. (But for now the defaults seem fine.)

Reply:
Decorating `handle_error` would only work in `handle_error` ran the actually HTTP request internally, like this:

```py
handle_error(httpx.get, "http://...")
```

Because it receives the request:

```py
handle_error(httpx.get("http://..."))
```

it's too late to retry.

Reply:
The decorator can only catch and retry exceptions arising from the call stack of the decorated function.

```py
In [1]: import httpx

In [2]: import stamina

In [3]: @stamina.retry(on=httpx.HTTPError)
   ...: def handle_error(response):
   ...:     ...
   ...:     return response.raise_for_error()
   ...:

In [4]: handle_error(httpx.get("http://localhost:9999"))  # ConnectionError
...
ConnectError: [Errno 111] Connection refused
```

Reply:
Ahhh I see. I'm also noticing there are some examples in `array` that don't use the retry directly on the `handle_error` call, so we can't just nest a `handle_error: with retries: handle_http_error`. 
Thanks for also answering some of my questions before I could even hit enter.